pocs/tests/test_scheduler.py

```diff
@@ -0,0 +1,32 @@
+import pytest
+
+from pocs.scheduler import create_scheduler_from_config
+from pocs.site_location_details import setup_site_location_details_from_config
+from pocs.utils import error
+
+
+def test_bad_scheduler_type(config):
+    conf = config.copy()
+    conf['scheduler']['type'] = 'foobar'
+    details = setup_site_location_details_from_config(config)
+    with pytest.raises(error.NotFound):
```

Comment:
This is actually a good example of how code coverage can help expose bugs even if the tests don't seem to. Looking at your PR I'm realizing we've had a problem in our scheduler testing, which is not something that you introduced but instead was already there.

If you look at the code coverage for the scheduler you'll see that there is one line that is not getting hit, namely the error that should be raised for an invalid scheduler type ([line 64](https://codecov.io/gh/panoptes/POCS/pull/838/diff?src=pr&el=tree#D5-64)). What should be happening is that the code should be unable to find the `foobar` scheduler (because it doesn't exist) and so it should raise a ` NotFound` exception on line 64. However the red is telling us that piece of code is never getting run, which means we are never actually using it.

What is going on is that this test, which should be checking for the bad scheduler type, is actually failing because of the missing fields path check on [line 37](https://github.com/panoptes/POCS/blob/70416431f23476a3b535513ff59ffc1057e6407c/pocs/scheduler/__init__.py#L37). (The fields file is checked in the next test below).  So, basically we are not really ever checking for a bad scheduler type.

And looking even deeper, the `load_module` function itself is [catching the ImportError](https://github.com/panoptes/POCS/blob/develop/pocs/utils/__init__.py#L316), and is raising the `error.NotFound`, which means the [ImportError on line 63](https://github.com/panoptes/POCS/blob/70416431f23476a3b535513ff59ffc1057e6407c/pocs/scheduler/__init__.py#L63) could never possibly get raised because `load_module` is re-raising the `ImportError` but as an `error.NotFound` instead.

Again, this was not something you have done but was something that was already there. However let's use this time to go ahead and clean it up.

Two things need to be done here:

* [Line 63](https://github.com/panoptes/POCS/blob/70416431f23476a3b535513ff59ffc1057e6407c/pocs/scheduler/__init__.py#L63) should change to be `except error.NotFound as e`. This is a little bit pointless as it just re-raises the same exception on the next line and so you could alternatively just remove the entire `try/except` block (lines 39 and 63). However let's keep it and just re-raise the `error.NotFound` for now. In the future we would maybe want to do something smarter there instead (such as create a `error.ImportError` of our own), so it makes sense to leave it in place.
* This test needs to have a valid `conf['scheduler']['fields_file']`. That is, it should be set to a real path name such that the existence check (line [37](https://github.com/panoptes/POCS/blob/70416431f23476a3b535513ff59ffc1057e6407c/pocs/scheduler/__init__.py#L37)) evaluates to True.

Once this is done the tests should pass, as they do now, but that [red marker on line 64](https://codecov.io/gh/panoptes/POCS/pull/838/diff?src=pr&el=tree#D5-64) should disappear and our coverage should go up (a tiny bit).

That's a pretty subtle bug. Let me know if anything doesn't make sense.  Thanks!

Reply:
But yes, this is a good example of why you should look at the coverage reports and try to figure out _why_ the red chunks are red.

Reply:
@wtgee Should I try to increase coverage in Observatory class?

Reply:
Left a message in slack a few days ago, but no, not unless it is directly related to this PR. We want to try to keep PRs focused.